benchmarks/causal_discovery.py

```diff
@@ -0,0 +1,111 @@
+import numpy as np
+from pgmpy.base import DAG
+from pgmpy.estimators import PC, GES
+from pgmpy.metrics import SHD
+from pgmpy.factors.continuous import LinearGaussianCPD
+from pgmpy.models import LinearGaussianBayesianNetwork as LGBN
+
+"""
+Benchmarking Structural Hamming Distance (SHD) for Causal Discovery Algorithms: PC and GES
+
+Algorithm Definitions:
+----------------------
+- PC (Peter-Clark) Algorithm:
+  A constraint-based algorithm that starts with a complete undirected graph and removes edges
+  based on conditional independence tests. It then orients edges using separation sets and
+  rules like the collider rule:
+      X → Z ← Y   if X ⟂⟂ Y | Z
+
+- GES (Greedy Equivalence Search) Algorithm:
+  A score-based algorithm that performs greedy forward and backward search in the space
+  of equivalence classes of DAGs to maximize a scoring criterion such as BIC.
+
+  Scoring function (Bayesian Information Criterion - BIC):
+      Score(G : D) = log P(D | G) - λ * |G|
+
+Metric:
+-------
+- SHD (Structural Hamming Distance):
+  Measures the number of edge insertions, deletions, or reversals required to convert
+  one DAG into another.
+"""
+
+def generate_random_dag(num_nodes: int, edge_prob: float = 0.3, seed: int = 0) -> DAG:
+    dag = DAG.get_random(n_nodes=num_nodes, edge_prob=edge_prob, seed=seed)
+    for i in range(num_nodes):
+        dag.add_node(f"X_{i}")
```

Comment:
Why do we need to add the nodes here? Doesn't the `get_random` method already give the DAG on the specified number of nodes?